File: documentation/manual/working/scalaGuide/main/akka/code/components/ComponentsWithClusterSharding.scala

package scalaguide.akka {

//#cluster-sharding
import play.api._
import play.api.ApplicationLoader.Context
import play.api.routing.Router
import play.scaladsl.cluster.sharding.typed.ClusterShardingComponents

class MyApplicationLoader extends ApplicationLoader {
  def load(context: Context) = {
    new MyComponents(context).application
  }
}

class ComponentsWithClusterSharding(context: Context) 
  extends BuiltInComponentsFromContext(context) 
    with ClusterShardingComponents {
      
  lazy val router = Router.empty
}
//#cluster-sharding
}

File: documentation/manual/working/javaGuide/main/akka/code/javaguide/akka/components/ComponentsWithClusterSharding.java
package javaguide.akka;

// #cluster-sharding
import play.Application;
import play.ApplicationLoader;
import play.BuiltInComponentsFromContext;
import play.controllers.AssetsComponents;
import play.routing.Router;
import play.javadsl.cluster.sharding.typed.ClusterShardingComponents;

public class ComponentsWithClusterSharding extends BuiltInComponentsFromContext
    implements ClusterShardingComponents, AssetsComponents {

  public ComponentsWithClusterSharding(ApplicationLoader.Context context) {
    super(context);
  }

  @Override
  public Router router() {
    return Router.empty();
  }
}
// #cluster-sharding
